app/controllers/api/migrations.php

```diff
@@ -290,6 +297,98 @@
+    ->label('audits.event', 'migration.create')
+    ->label('sdk', new Method(
+        namespace: 'migrations',
+        name: 'createCsvMigration',
+        description: '/docs/references/migrations/migration-csv.md',
+        auth: [AuthType::ADMIN],
+        responses: [
+            new SDKResponse(
+                code: Response::STATUS_CODE_ACCEPTED,
+                model: Response::MODEL_MIGRATION,
+            )
+        ]
+    ))
+    ->param('bucketId', '', new UID(), 'Storage bucket unique ID. You can create a new storage bucket using the Storage service [server integration](https://appwrite.io/docs/server/storage#createBucket).')
+    ->param('fileId', '', new UID(), 'File ID.')
+    ->param('resourceId', null, new CompoundUID(), 'Composite ID in the format {databaseId:collectionId}, identifying a collection within a database.')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('project')
+    ->inject('deviceForFiles')
+    ->inject('deviceForCsvImports')
+    ->inject('queueForEvents')
+    ->inject('queueForMigrations')
+    ->action(function (string $bucketId, string $fileId, string $resourceId, Response $response, Database $dbForProject, Document $project, Device $deviceForFiles, Device $deviceForCsvImports, Event $queueForEvents, Migration $queueForMigrations) {
+        $isAPIKey = Auth::isAppUser(Authorization::getRoles());
+        $isPrivilegedUser = Auth::isPrivilegedUser(Authorization::getRoles());
+
+        $bucket = Authorization::skip(fn () => $dbForProject->getDocument('buckets', $bucketId));
+
+        if ($bucket->isEmpty() || (!$isAPIKey && !$isPrivilegedUser)) {
+            throw new Exception(Exception::STORAGE_BUCKET_NOT_FOUND);
+        }
+
+        $file = Authorization::skip(fn () => $dbForProject->getDocument('bucket_' . $bucket->getInternalId(), $fileId));
+        if ($file->isEmpty()) {
+            throw new Exception(Exception::STORAGE_FILE_NOT_FOUND);
+        }
+
+        $path = $file->getAttribute('path', '');
+        if (!$deviceForFiles->exists($path)) {
+            throw new Exception(Exception::STORAGE_FILE_NOT_FOUND, 'File not found in ' . $path);
+        }
+
+        if (!empty($file->getAttribute('openSSLCipher')) || $file->getAttribute('algorithm', Compression::NONE) !== Compression::NONE) {
+            throw new Exception(Exception::STORAGE_FILE_TYPE_UNSUPPORTED, "Only uncompressed, unencrypted CSV files can be used for document import.");
+        }
+
+        // copy to temporary folder
+        $migrationId = ID::unique();
+        $newPath = $deviceForCsvImports->getPath('/' . $migrationId . '_' . $fileId . '.csv');
+        if (!$deviceForFiles->transfer($path, $newPath, $deviceForCsvImports)) {
+            throw new \Exception("Unable to copy file");
+        }
+
+        $fileSize = $deviceForCsvImports->getFileSize($path);
+        $resources = Transfer::extractServices([Transfer::GROUP_DATABASES]);
```

Comment:
I think this should only be `[Resource::TYPE_DOCUMENT]`

Reply:
No, we need a `root resource`.
See [here](https://github.com/utopia-php/migration/blob/main/src/Migration/Transfer.php#L235) and [here](https://github.com/utopia-php/migration/blob/main/src/Migration/Transfer.php#L316C5-L332C6). We seem to be revolving more around `ROOT_*` on the public front of the library.